### PR TITLE
fix: use zod for validation

### LINKED ISSUE #5475: [CAL-349] In edit team page, team URL is not validated appropriately
Found a bug? Please fill out the sections below. 👍

### Issue Summary

In edit team page, team URL is not validated appropriately.
It looks like anything can be input and save

### Steps to Reproduce

1. Go to `/teams` page, Select one team for editing
2. In `/settings/teams/{selectedTeamID}/profile` page, Input something in `Team URL` field.
3. Click `Update` button.
4. Check the saved Team URL.

In my test case, it could be updating with not percent encoding characters. (like <, >, +, etc...)

Furthermore, I think when create a team, Non-english characters are need to be encoded with percent encoding. Currently, It's just replace with `-`.

### Technical details

* Chrome 107.0.5304.87 (arm64)
* Node.js 16.x, git cloned current release (v2.1.4)

> From [SyncLinear.com](https://synclinear.com)

### PR DESCRIPTION
**Pull request by Udit-takkar**
Fixes: #5475

### CODE REVIEW

**Comment by Udit-takkar** [comment]
https://user-images.githubusercontent.com/53316345/201669813-5f39e1f1-1e9f-45ff-8e9c-fff5a99fe3fb.mov

File: packages/features/ee/teams/pages/team-profile-view.tsx [hunk 1]
```diff
@@ -19,12 +21,19 @@ import Meta from "@calcom/ui/v2/core/Meta";
...
+const regex = new RegExp("^[a-zA-Z0-9-]*$");
+
+const teamProfileFormSchema = z.object({
+  name: z.string(),
+  url: z
+    .string()
+    .regex(regex, {
+      message: "Url can only have alphanumeric characters(a-z, 0-9) and hyphen(-) symbol.",
+    })
+    .min(1, { message: "Url cannot be left empty" }),
```
Comment (Udit-takkar) [inline packages/features/ee/teams/pages/team-profile-view.tsx:33]: should we also add any maximum length ?

**Review by emrysal — DISMISSED** [review]

File: packages/features/ee/teams/pages/team-profile-view.tsx [hunk 2]
```diff
@@ -19,12 +21,19 @@ import Meta from "@calcom/ui/v2/core/Meta";
...
-  logo: string;
-  bio: string;
-}
+const regex = new RegExp("^[a-zA-Z0-9-]*$");
+
+const teamProfileFormSchema = z.object({
+  name: z.string(),
+  url: z
+    .string()
+    .regex(regex, {
```
[the remark this answers is not in the document]
Comment (Udit-takkar) [reply packages/features/ee/teams/pages/team-profile-view.tsx:30]: i think z.url() checks for a  valid url ( https:// ) while here url is just url slug for team

**Review by zomars — COMMENTED** [review]
Thanks @Udit-takkar let's just address the feedback first before merging 🙏🏽

File: packages/features/ee/teams/pages/team-profile-view.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (zomars) [reply packages/features/ee/teams/pages/team-profile-view.tsx:30]: @Udit-takkar do we need to update this field name in the backend as well? Just let's make sure that is still working properly

**Review by zomars — APPROVED** [review]
LGTM

**Pull request merged**